### DISCUSSION TITLE
Are reference-style links supported in python docstrings?

### DISCUSSION THREAD

**Discussion by cdwilson** [mkdocstrings/python#178] [Q&A] [answered]
Is it possible to use [reference-style links](https://www.markdownguide.org/basic-syntax/#reference-style-links) in a python docstring when using mkdocstrings?

I tried the following example:

```python
def function_with_types_in_docstring(param1, param2):
    """Example function with types documented in the docstring.

    [PEP 484](https://www.python.org/dev/peps/pep-0484/) type
    annotations are supported. If attribute, parameter, and return types
    are annotated according to [PEP 484][1], they do not need to be
    included in the docstring:

    Args:
        param1 (int): The first parameter.
        param2 (str): The second parameter.

    Returns:
        (bool): The return value. True for success, False otherwise.

    [1]: https://www.python.org/dev/peps/pep-0484/
    """
```

The first link (`[PEP 484](https://www.python.org/dev/peps/pep-0484/)`) is rendered correctly, but the second reference-style link (`[PEP 484][1]`) is not:

<img width="706" alt="image" src="https://github.com/user-attachments/assets/c1efd2c8-81ee-43b8-af84-515501fb059e">

**Comment by pawamoy** [comment, marked as answer]
Since this is a Google-style docstring, and I suppose mkdocstrings-python/Griffe was configured to parse it as such, the blocks of text are converted from Markdown to HTML separately, which explains why the reference `[1]` fails.

![codesnippet](https://github.com/user-attachments/assets/8f697135-400b-45ff-85e2-0d4e42626fce)

To fix this, you must put the reference link in the same text block as where it's referenced:

```python
def function_with_types_in_docstring(param1, param2):
    """Example function with types documented in the docstring.

    [PEP 484](https://www.python.org/dev/peps/pep-0484/) type
    annotations are supported. If attribute, parameter, and return types
    are annotated according to [PEP 484][1], they do not need to be
    included in the docstring:

    [1]: https://www.python.org/dev/peps/pep-0484/

    Args:
        param1 (int): The first parameter.
        param2 (str): The second parameter.

    Returns:
        (bool): The return value. True for success, False otherwise.
    """
```

**Reply by cdwilson** [reply]
Got it, I didn't realize that the blocks were processed independently. As you suggest, moving the reference into the same block works. Thanks @pawamoy!
